sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -4257,6 +4257,27 @@ def test_workspace_config_validation(self):
             workspace.set_size('   ')
         self.assertIn('required and cannot be empty', str(context.exception))
 
+    def test_compile_fails_when_workspace_config_has_empty_size(self):
+        """Test that compilation should fail if workspace is configured without appropriate size."""
+        empty_sizes = ['', '   ', '\t', '\n', '  \t  \n  ']
+        for size in empty_sizes:
+            with self.subTest(size=repr(size)):
+                workspace_config = WorkspaceConfig(size='10Gi')
+                workspace_config.size = size
+                config = PipelineConfig(workspace=workspace_config)
```

Comment:
Could you add a test for when workspace is never set in the pipeline config? I imagine that will be the most common mistake.